HIP/hip-1056.md

```diff
@@ -0,0 +1,1933 @@
+are sent) was considered. In this case every entity could have a last updated counter that is updated on entity changes, if the
+value is over a certain threshold then the whole entity is externalized, if not only deltas would be externalized. This would optimize
+how far back a client would have to go to retrieve state info. However, this approach would require the counter to be stored in state
+for every entity which has undesired impacts on state size.
+Another inspired approach was that on the first observation of an entity in a block the CN would externalize the full entity state,
+after that only the delta changes would be externalized for the remainder of the block.
+In this way any consumer of a block would have the complete state by applying the diffs onto the initial state.
+This shares the downside of requiring additional temporary storage, and also requires the creation of a "difference" form for every
+entity potentially stored in state, and could, paradoxically, _increase_ the total size of blocks with very many changes to small entities.
+Upon early POC considerations of compression the state size gains of the block stream size when compared to the record v6 externalized
+stream mitigated the concerns and thus the explored solutions were not needed. 
+
+### Including exchange_rate in the TransactionResult
+In the past the exchange_rate details were in the transaction record. However, this information doesn't change often
+and is based on the value of a system file. Changes to this file are externalized in the block stream, thus the current
+exchange rate value is always available via a MN (and future BN). Thus it is no longer necessary to store the
+`exchange_rate` in the stream and we can remove unnecessary data from the block stream.
+
+### Changing `EthereumOutput` to `EVMOutput`
+`EthereumOutput` seemed more appropriate if called `EVMOutput`. However, this is a legacy item referring ot the
+`EthereumTransaction` input whose name can't be changed for backwards compatibility. Since outputs in the Block Stream
+must match the transaction input it would nopt be easy to change it. Additionally, there's a `ContractCall`,
+`ContractCreate` and `EthereumTransaction`, `EVMOutput` would actually apply to all 3 and is thus too general.
+
+## Open Issues
+Open Issues captured here and in https://github.com/hashgraph/hedera-block-node/issues/37
+
+- [ ] Q: Are there additional protocol buffer changes needed to reduce the
+      stream overhead for Events, Transactions, or other frequent entities.
+   - Transaction rearrangement - utilize new wrapper and remove old references
+   - Potentially move previous block has from block header to block proof
+   - Update smart contract output as its missing details
+   - Optimizations to reduce number of message objects
+   - Ensure we are not double nesting in Transaction result vs Transaction output
```

Comment:
We should include the "how to handle `File` content for large files" here, as that's an open question that may have somewhat significant impact (particularly w.r.t. "upgrade" files).